packages/crayons-core/src/components/select/select.tsx

```diff
@@ -345,6 +345,22 @@ export class Select {
     this.renderInput();
   }
 
+  @Watch('options')
+  onOptionsChange(newValue) {
+    this.dataSource = newValue;
+    const selectedValues = newValue?.filter((option) => option.selected);
```

Comment:
This seems to be redundant in componentWillLoad too. Should we separate this out into a private function?

Reply:
During componentWillLoad the datasource can be either the slot or the option prop, but in case of `@Watch('options')` the datasource is just options and we follow the hierarchy here datasource > value > selectedOptions but at this point setting the value before options wont work because on componentWillLoad this hierarchy is not there.